api/v1/scheduledpodautoscaler_types.go

```diff
@@ -0,0 +1,91 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ScheduledPodAutoscalerSpec defines the desired state of ScheduledPodAutoscaler
+type ScheduledPodAutoscalerSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Resource field for ScheduledPodAutoscaler - the resource to scale:
+	// Requires two fields - name and type:
+	Resource Resource `json:"resource"`
+
+	// Setup for ScaleUp filed
+	// Includes two fields - time and value:
+	ScaleUp ScaleSpec `json:"scaleUp"`
```

Comment:
I think this can be simplified to events as in reality the scaleUp and scaleDown are doing the exact same thing, setting the desired value to a value at a specified time.

ie:
```yaml
scheduledEvents:
- name: Event1
  time: 8:00AM
  value: 25
- name: Event2
  time: 6:00PM
  value: 5
```

This could also be extended in the future to add days of week as well

Reply:
Spot-on @andrewhertog, you're correct there's no real scaleUp or scaleDown setup (as its mainly based on the value). Were you thinking multiple events should occur (beyond just 2) hence having a list?